### PR TITLE
Clang tidy run on complete demo project (no cuda code)

### PR DESCRIPTION
**Pull request by TheCodez**
@cbachhuber I haven't been able to include CUDA files in this.

### CODE REVIEW

File: .github/workflows/ubuntu_build.yml [hunk 1]
```diff
@@ -33,4 +33,4 @@ jobs:
     - name: Install clang-tidy
       run: sudo apt install clang-tidy
     - name: Clang-Tidy
-      run: find dogm/demo/utils -iname '*.cpp' | xargs clang-tidy -p dogm/build
+      run: find dogm/demo -iname '*.cpp' | xargs clang-tidy -p dogm/build
```
Comment (cbachhuber) [inline .github/workflows/ubuntu_build.yml:36]: Very cool so far! To check CUDA, how about
```suggestion
      run: find dogm -iname '*.cpp' -iname '*.cu' | xargs clang-tidy -p dogm/build
```

Though I'm not sure how many errors are lurking here and if clang-tidy is able to check these (cope with CUDA dependencies).
I planned to do that in a separate PR, but you can certainly go ahead and do it in this one :)

Comment (TheCodez) [reply .github/workflows/ubuntu_build.yml:36]: I already tried this and it results in the following error:
`Error while processing DOGM/dogm/demo/simulator/mapping/kernel/measurement_grid.cu.`

Comment (cbachhuber) [reply .github/workflows/ubuntu_build.yml:36]: And no more info?

[It looks like](https://stackoverflow.com/questions/47361538/use-clang-tidy-on-cuda-source-files) there are more things to consider when running clang-tidy on CUDA sources. You can try that out, or alternatively, I would play around with this in the coming days.

Comment (TheCodez) [reply .github/workflows/ubuntu_build.yml:36]: If you want you can play around with it 👍 I'll merge this pr as is then.
